firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentReference.java

```diff
@@ -59,7 +58,7 @@
  * in test mocks. Subclassing is not supported in production code and new SDK releases may break
  * code that does so.
  */
-@Serializable(with = FirestoreNativeDataTypeSerializer.DocumentReferenceSerializer.class)
+@Serializable(with = DocumentRefContextualSerializer.class)
```

Comment:
`DocumentRefContextualSerializer` this is the Java serializer. Note that the only kotlin files in Java lib are the annotation files.